### PR TITLE
handle two corner cases where preload would get called twice

### PR DESCRIPTION
**Pull request by ZappoMan**
Fixes and issue that we were seeing in cell science content with preload being called multiple times.

### CODE REVIEW

File: libraries/entities-renderer/src/EntityTreeRenderer.cpp [hunk 1]
```diff
@@ -690,10 +690,11 @@ void EntityTreeRenderer::entitySciptChanging(const EntityItemID& entityID, const
 void EntityTreeRenderer::checkAndCallPreload(const EntityItemID& entityID, const bool reload) {
     if (_tree && !_shuttingDown) {
         EntityItemPointer entity = getTree()->findEntityByEntityItemID(entityID);
-        if (entity && !entity->getScript().isEmpty()) {
+        if (entity && entity->shouldPreloadScript()) {
```
Comment (ZappoMan) [inline libraries/entities-renderer/src/EntityTreeRenderer.cpp]: ultimately, the entity is the authority on whether or not the desired script is different from the know "loaded" script. so we move that logic into the entity.

**Comment by hifi-gustavo** [comment]
Build console output for commit: a40a562db21152bd94a393b2226d5586b8606c04 available at: https://gist.github.com/d7c846563bfbc4880855<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7083.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7083.dmg

**Comment by sethalves** [comment]
test this please

**Comment by ZappoMan** [comment]
this fixes this reported bug: https://app.asana.com/0/26320649018758/84176462198297

**Comment by hifi-gustavo** [comment]
Build console output for commit: a40a562db21152bd94a393b2226d5586b8606c04 available at: https://gist.github.com/d7c846563bfbc4880855<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7083.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7083.dmg

**Comment by ZappoMan** [comment]
Let's make sure @imgntn does some testing on this... since he requested this bug fix.

**Comment by imgntn** [comment]
working well for me on windows 10 when switching domains -- seeing appropriate preloads without extras

**Pull request merged**